### DISCUSSION TITLE
Extension clones entire Desktop folder into AppData. Is this inteded?

### DISCUSSION THREAD

**Discussion by mitkan191003** [redhat-developer/vscode-java#2167] [Q&A] [answered]
Hi,

The Redhat Java extension seems to have cloned my entire C:\Users\\[me]\Desktop directory into my AppData folder, 17GB in total. Is this intended and is there a way to turn it off?
![image](https://user-images.githubusercontent.com/76726314/137423742-ddee3132-2310-47bf-92e1-2bffe102179c.png)

**Comment by 0dinD** [comment]
Those are probably index files generated by the compiler, see #2110 which tracks the issue of automatically cleaning these up (also you can safely delete them).

It's usually not so bad unless you have huge projects, but from your screenshot it looks like you've opened your entire Desktop folder as a Java project. Is that what you intended? Normally you'd open one project at a time and only the index files for that project would be generated.

**Reply by mitkan191003** [reply]
They aren't index files. If I go into the folders, I can find, for example, mp3 files that will playback if double-clicked on. 
![image](https://user-images.githubusercontent.com/76726314/137822184-1836c5c6-4154-44ac-a766-3e7303c7a20e.png)
![image](https://user-images.githubusercontent.com/76726314/137822238-61e2503e-d89f-4820-b6ce-749e4ceed33c.png)



As for opening my desktop as a project, I have test files such as scrap.java that I use to test code. I just keep it on my desktop for easy access. In VS Code I open my Desktop in the explorer sidebar, but I had no idea that it would affect the java compiler. Should I just avoid doing so in the future or should I create a folder to store the test file in?

**Reply by 0dinD** [reply]
Oh, I think I see what's happened: You've added the entire Desktop folder as a source path via `java.project.sourcePaths`, right? This means that all your files in the Desktop folder are treated as either Java source files or project resources that need to be included with the compiled source files (in the `bin/` folder you see in your `workspaceStorage`). If you compiled your project into a `.jar` file, you'd want those resource files to be included (or else you would not put them on your source path).

So what you should do is right click the Desktop folder in the VS Code explorer, and select **Remove Folder from Java Source Path**. Then, you can delete the duplicate files in your `workspaceStorage` and they shouldn't be copied there again. If you want to have some source files included on the source path, you'll need to make a separate directory (although it could still be inside the Desktop folder) for them and right click + select **Add Folder to Java Source Path**.

**Reply by mitkan191003** [reply]
I actually haven't tinkered with the extension settings at all. 
![image](https://user-images.githubusercontent.com/76726314/138191990-feecfaac-238f-412e-8484-cabee4d11a34.png)

**Reply by 0dinD** [reply]
So when you open a Java file from your Desktop folder, does it show a warning at the top telling you that it's not on your classpath (which is what you're after in this case)? If not, have you tried right click + **Remove Folder from Java Source Path**?

**Reply by mitkan191003** [reply]
I don't see any warning or error squiggles. Not sure if this is what you're talking about. I just want to test code in VS Code. I'm not interested in compiling them into .jars so I never bothered with project files.
![image](https://user-images.githubusercontent.com/76726314/138210239-3aa8f447-7230-463a-bfa1-7987427398f5.png)
![image](https://user-images.githubusercontent.com/76726314/138210793-c18303fa-33dd-40bb-9116-5083735c4cad.png)
![image](https://user-images.githubusercontent.com/76726314/138210706-65a053c6-75e6-4e31-8ab9-377704a0bc86.png)

**Comment by jdneo** [comment, marked as answer]
@mitkan191003 You can first open a Java file to activate the Java extension. Then those menu items can be seen in the context menu.

Another way to do it is trigger the command `Java: Configure Classpath` to show a webview UI to do the same thing.
